app/src/main/java/com/github/sdp/ratemyepfl/database/ItemRepositoryImpl.kt

```diff
@@ -0,0 +1,37 @@
+package com.github.sdp.ratemyepfl.database
+
+import com.github.sdp.ratemyepfl.model.items.Reviewable
+import com.google.firebase.firestore.DocumentSnapshot
+
+/**
+ * Base class for any item repository. An item is any object that is reviewable.
+ *
+ * @param collectionPath: Specify the path to the Firebase collection
+ *
+ * Note: For clean dependency injection, each subclass should provide a binding as
+ * done in RepositoryModule. When you want to inject an ItemRepository, you should
+ * use ItemRepository<YourClass> instead of YourClass.
+ */
+sealed class ItemRepositoryImpl<T : Reviewable> constructor(collectionPath: String) :
+    ItemRepository<T>, Repository(collectionPath) {
+
+    companion object {
+        const val DEFAULT_LIMIT: Long = 50
```

Comment:
Do we really need to keep the limit ?

Reply:
I don't know since I did not manage both the databases and the repositories implementation. I'll let it there for now.